src/diffusers/modeling_utils.py

```diff
@@ -124,10 +125,42 @@ class ModelMixin(torch.nn.Module):
     """
     config_name = CONFIG_NAME
     _automatically_saved_args = ["_diffusers_version", "_class_name", "_name_or_path"]
+    supports_gradient_checkpointing = False
 
     def __init__(self):
         super().__init__()
 
+    @property
+    def is_gradient_checkpointing(self) -> bool:
+        """
+        Whether gradient checkpointing is activated for this model or not.
+
+        Note that in other frameworks this feature can be referred to as "activation checkpointing" or "checkpoint
+        activations".
+        """
+        return any(hasattr(m, "gradient_checkpointing") and m.gradient_checkpointing for m in self.modules())
+
+    def gradient_checkpointing_enable(self):
```

Comment:
```suggestion
    def enable_gradient_checkpointing(self):
```

Now that we have the enable_* pattern with `enable_attention_slicing` :sweat_smile: